src/ast/nodes/IfStatement.ts

```diff
@@ -90,14 +88,22 @@ export default class IfStatement extends StatementBase implements DeoptimizableE
 		}
 	}
 
+	private getTestValue(): LiteralValueOrUnknown {
```

Comment:
The changes in this file make sense but I do not think they actually solve any issue; on the other hand, they add a lot of additional function calls in various places. Usually, the deoptimization logic should fix it if the test value is later reassigned. The reason why it was not working is because there is a bug in `MemberExpression`: 
`deoptimizeCache` https://github.com/rollup/rollup/blob/f9b9a7cb12d9945cd49866fb6af6a74994665ff4/src/ast/nodes/MemberExpression.ts#L114-L118 should not only deoptimize all dependent `expressionsToBeDeoptimized` but also set `this.propertyKey = UnknownKey`. With that change, I can revert all changes in `IfStatement` and still have all tests green.

So how do `DeoptimizeableEntity`s work?  This is a performance optimization that allows nodes to cache values (such as `this.testValue`) reliably while still making sure the value is reset if one of the entities needed to get the cached value is reassigned. To do that, `getLiteralValueAtPath` receives the calling Node as parameter. Then when the value that was initially returned is deoptimized, `deoptimizeCache` is called on the caller to make it "forget" the cached value. This was missing for `MemberExpression`. A similar logic is in place for caching function return values.

Reply:
`MemberExpression` is fixed now and I think the changes are ok as I think moving logic away from `bind` is a good thing.